core/src/main/java/nu/yona/server/subscriptions/service/UserService.java

```diff
@@ -230,4 +257,63 @@ public void addBuddy(UserDTO user, BuddyDTO buddy)
 		userEntity.addBuddy(buddyEntity);
 		User.getRepository().save(userEntity);
 	}
+
+	@Transactional
+	public NewDeviceRequestDTO setNewDeviceRequestForUser(UUID id, String userPassword, String userSecret)
+	{
+		User userEntity = getEntityByID(id);
+		Date expirationDateTime = getNewDeviceRequestExpirationDateTime();
+		NewDeviceRequest newDeviceRequestEntity = NewDeviceRequest.createInstance(userPassword, expirationDateTime);
+		newDeviceRequestEntity.encryptUserPassword(userSecret);
+		userEntity.setNewDeviceRequest(newDeviceRequestEntity);
+		return NewDeviceRequestDTO.createInstance(User.getRepository().save(userEntity).getNewDeviceRequest());
+	}
+
+	private Date getNewDeviceRequestExpirationDateTime()
+	{
+		Date now = new Date();
+		Calendar c = Calendar.getInstance();
+		c.setTime(now);
+		c.add(Calendar.DATE, 1);
+		Date expirationDateTime = c.getTime();
+		return expirationDateTime;
+	}
+
+	@Transactional
+	public NewDeviceRequestDTO getNewDeviceRequestForUser(UUID id, String userSecret)
+	{
+		if (StringUtils.isNotBlank(userSecret))
```

Comment:
make this ``isNotBlank`` and swap the if-else. Positive conditions read a lot easier.